unlock-app/src/components/creator/CreatorLockForm.js

```diff
@@ -36,31 +48,67 @@ export class CreatorLockForm extends React.Component {
       name: props.name,
       address: props.address,
     }
-    this.state.valid = {
-      name: this.validate('name', props.name),
-      expirationDuration: this.validate(
-        'expirationDuration',
-        props.expirationDuration
-      ),
-      keyPrice: this.validate('keyPrice', props.keyPrice),
-      maxNumberOfKeys: this.validate('maxNumberOfKeys', props.maxNumberOfKeys),
-    }
-
+    const { validityState: valid, errors } = this.valid(this.state)
+    this.state.valid = valid
+    this.state.errors = errors
     this.handleSubmit = this.handleSubmit.bind(this)
     this.handleCancel = this.handleCancel.bind(this)
     this.handleChange = this.handleChange.bind(this)
     this.handleUnlimitedClick = this.handleUnlimitedClick.bind(this)
+    this.saveLock = this.saveLock.bind(this)
+    this.processFormErrors = this.processFormErrors.bind(this)
+  }
+
+  valid(state) {
+    // the list of errors we will pass to setError
+    const errors = []
+    // for each field, retrieve the error triggered by current state
+    // and then make sure we set it as existing.
+    const validityState = [
+      'expirationDuration',
+      //'expirationDurationUnit',
+      'keyPrice',
+      //'keyPriceCurrency',
+      'maxNumberOfKeys',
+      'name',
+    ].reduce((fieldValidity, field) => {
```

Comment:
nit: would `formValidity` be a better name to indicate that this is about all the fields?